### PR TITLE
BEMXJST: generate() should support arrow functions

### PR DESCRIPTION
**Pull request by miripiruni**
Fix for #259 

So, the checks have failed [because of Node v0.1 and Node v0.12](https://travis-ci.org/bem/bem-xjst/builds/131134175). 

WIP: I would like to disable ES6 tests for old Node.js. 

cc @zxqfox

### CODE REVIEW

File: lib/compiler.js [hunk 1]
```diff
@@ -15,8 +15,14 @@ Compiler.prototype.generate = function generate(code, options) {
...
   // Yeah, let people pass functions to us!
-  if (typeof code === 'function')
-    code = code.toString().replace(/^function\s*\w*\(\)\s*{|}$/g, '');
+  if (typeof code === 'function') {
+    code = code.toString();
+    code = code.replace(code.indexOf('function') === 0 ?
+      // Regular function
+      /^function\s*\w*\(\)\s*{|}$/g :
+      // Arrow function
+      /^\(\)\s=>\s{|}$/g, '');
```
Comment (zxqfox) [inline lib/compiler.js]: Please add support for `_ => {}`.

Hm, we don't have variables at all in the `function(){ … }` variant.

Btw, there could be spaces between parens. We could switch to `function\s*[^{]+{…` and `(\w|[^=>]+)\s*=>\s*\{` — these ones are relaxed variants for the first one, they could be much more useful for demo.

**Comment by miripiruni** [comment]
@zxqfox BTW, should I declare that bem-xjst need node >= 4.x? (Not v0.12 or lower)

**Comment by zxqfox** [comment]
@miripiruni It's not required, actually, to support both syntaxes, isn't it?

**Comment by miripiruni** [comment]
@zxqfox yep. In this branch I can turn off node 0.12/0.1 on Travis.

**Comment by zxqfox** [comment]
I don't think it's needed for this issue. So 👎 for dropping 0.10, 0.12.

**Comment by miripiruni** [comment]
@zxqfox I would like to disable ES6 tests for old Node.js.

File: test/modes-elemmod-test.js [hunk 2]
```diff
@@ -17,8 +17,9 @@ describe('Modes elemMod(elemModName, elemModVal)', function() {
     },
     {
       block: 'b',
-      content: { elem: 'inner',
-      elemMods: { valid: true }
+      content: {
+        elem: 'inner',
+        elemMods: { valid: true }
```
Comment (zxqfox) [inline test/modes-elemmod-test.js]: Why this change?

Comment (miripiruni) [inline test/modes-elemmod-test.js]: @zxqfox because of update JSCS. Now JSCS are report about this.

Comment (zxqfox) [inline test/modes-elemmod-test.js]: Ah, my bad.

But can we split JSCS update and support for arrow functions?

**Comment by zxqfox** [comment]
There are two options I see:
- emulate arrow functions behaviour with `var arrowFunction = function() {}; arrowFunction.toString = function(){ return "() => { block('a').tag()('a'); }"; }` — I don't see a big deal with tests for one regex.
- move ES6 code to separate file and skip the whole describe for nodes < 4:
  ```js
  (parseFloat(process.versions.node) < 4 ? describe : describe.skip)('ES6 pieces', function() {
    it('arrow functions', ...);
  });
  ```

**Comment by miripiruni** [comment]
@zxqfox second variant should not work because of SyntaxError with parse `() => {… }` in Node.js v0.12/v0.10.

**Comment by miripiruni** [comment]
Updated!

**Comment by zxqfox** [comment]
The second variant should not require files with ES6 syntax so it will work.

**Comment by zxqfox** [comment]
LGTM <3
